ironfish-cli/src/commands/wallet/chainport/send.ts

```diff
@@ -0,0 +1,421 @@
+    }
+
+    const transactionUrl = getExplorer(networkId)?.getTransactionUrl(hash)
+
+    if (transactionUrl) {
+      this.log(`\nBlock explorer: ${transactionUrl}`)
+    }
+
+    if (flags.watch) {
+      this.log('')
+
+      await watchTransaction({
+        client,
+        logger: this.logger,
+        account: from,
+        hash,
+      })
+    } else {
+      this.log(`Run ironfish wallet:transaction ${hash} to get the status of your transaction.`)
+      this.log(
+        `Run ironfish wallet:transaction:watch ${hash} to wait for the status of your transaction to be confirmed/ expired.`,
+      )
+    }
+  }
+
+  private async getAndValidateInputs(client: RpcClient, networkId: number) {
+    const { flags } = await this.parse(BridgeCommand)
+
+    let from = flags.account?.trim()
+    let to = flags.to?.trim()
+    let assetId = flags.assetId
+
+    if (!from) {
+      const response = await client.wallet.getDefaultAccount()
+
+      if (!response.content.account) {
+        this.error(
+          `No account is currently active.
+            Use ironfish wallet:create <name> to first create an account`,
+        )
+      }
+
+      from = response.content.account.name
+    }
+
+    if (!to) {
+      to = await CliUx.ux.prompt('Enter the public address of the recipient', {
+        required: true,
+      })
+    }
+
+    if (!validator.isAddress(to)) {
+      this.error('Invalid to ethereum address')
+    }
+
+    if (flags.expiration !== undefined && flags.expiration < 0) {
+      this.error('Expiration sequence must be non-negative')
+    }
+
+    const tokens = await fetchChainportVerifiedTokens(networkId)
+
+    if (assetId == null) {
+      const asset = await selectAsset(client, from, {
+        action: 'send',
+        showNativeAsset: true,
+        showNonCreatorAsset: true,
+        showSingleAssetChoice: false,
+        filter: (asset) => {
+          return tokens.some((t) => t.web3_address === asset.id)
+        },
+      })
+
+      assetId = asset?.id
+
+      if (!assetId) {
+        assetId = Asset.nativeId().toString('hex')
+      }
+    }
+
+    const asset: ChainportVerifiedToken | undefined = tokens.find(
+      (t) => t.web3_address === assetId,
+    )
+
+    if (!asset) {
+      const names = tokens.map(
+        (t, index) => `${index + 1}. ${t.name} (${t.symbol}) - ${t.web3_address}`,
+      )
+
+      this.error(
+        `Asset ${assetId} not supported by Chainport. Here are the supported tokens: \n\n${names.join(
+          '\n',
+        )}\n`,
+      )
+    }
+
+    const targetNetworks = asset.target_networks
+
+    const assetData = (
+      await client.wallet.getAsset({
+        account: from,
+        id: assetId,
+      })
+    ).content
+
+    if (assetData.verification.status === 'unverified') {
+      assetData.verification.decimals = asset.decimals
+      assetData.verification.symbol = asset.symbol
+      assetData.verification.status = 'verified'
+    }
```

Comment:
Is this because tokens could be in the Chainport verified token list, but not in our verified token list? Is it possible that decimals or symbol could be different between the two lists?

Reply:
Yes, that is why I added it. 

I guess it is possible. Chainport could send us an incorrect decimal place but I'm assuming that the chance of that is very low, but we would default to the decimals in the verified token list rather than the one provided by chainport. 

Not sure if there is a better approach to handle this case, lmk what you think.

Reply:
I'm not sure which decimals we should use if there's a mismatch 🤔 

I think it makes sense to treat our own list as the source of truth, but it wouldn't be good for us to display values one way and then have Chainport interpret the values differently on the other side

We should try to resolve this, but we can do that separately